src/Core/UbuntuDriversBackend.vala

```diff
@@ -66,23 +69,28 @@ public class AppCenterCore.UbuntuDriversBackend : Backend, Object {
                 continue;
             }
 
-            var driver_component = new AppStream.Component ();
-            driver_component.set_kind (AppStream.ComponentKind.DRIVER);
-            driver_component.set_pkgnames ({ package_name });
-            driver_component.set_id (package_name);
+            unowned string? nvidia_version = null;
+
+            if (package_name.has_prefix ("nvidia-driver-")) {
+                nvidia_version = package_name.offset (14);
+            } else if (package_name.has_prefix ("nvidia-")) {
+                nvidia_version = package_name.offset (7);
+            }
 
-            var icon = new AppStream.Icon ();
-            icon.set_name ("application-x-firmware");
-            icon.set_kind (AppStream.IconKind.STOCK);
-            driver_component.add_icon (icon);
+            if (null != nvidia_version) {
+                if (-1 == GLib.strcmp (latest_nvidia_version, nvidia_version)) {
```

Comment:
I know it's unlikely to happen anytime soon, but this will break when the driver version number grows from three to four digits. I think it would be cleaner to convert the number to an integer, and compare that.